### ISSUE TITLE
Errata - wrong link

### ISSUE THREAD

**Issue by 0b1001101** [djon2003/DNS-323#8]
Please accept my apologies for the incorrect link https://archive.debian.org/debian/dists/wheezy/main/installer-armel/current/images/orion5x/network-console/d-link/dns-323/netboot.img I shared before, as the image linked to it is for Wheezy; I would like to note that it is not a serious issue, as it works properly (this was my small project on a Sunday with a DNS323 that was left unused) and can be smoothly upgraded to Jessie once installed. The correct image is https://archive.debian.org/debian/dists/jessie/main/installer-armel/current/images/orion5x/network-console/d-link/dns-323/netboot.img

Furthermore, a mention of the Freexian Extended LTS project (which supports Jessie on armel) could be highly beneficial, as it will provide security updates et al. for our little device until 2025 https://www.freexian.com/lts/extended/docs/how-to-use-extended-lts/

**Comment by djon2003** [comment]
I changed (again... hihihi... I too, didn't realized the mistake) the URL to the net boot image. Thank you for reporting that.

Did you try the Freexian Extended LTS APT repositories?

**Comment by djon2003** [comment]
By the way, I presume you followed my instructions? How did you find them? Apart from your previous comments, do you have anything else?

**Comment by 0b1001101** [comment]
Soz again, mate: I feel like a drongo for my stuff-up :)

Now, I did follow (most of) your instruction and I found them to be top notch, although I first installed Wheezy (well, I had that image loaded and running and there’s no substantial difference in the installation process) and subsequently upgraded to Jessie. More on that later, anyway.

Yes, I do use the Freexian ELTS repository as, aside from their backports, it contains a mirror of archive.debian.org with its own signature - gone is the _W: GPG error: http://archive.debian.org jessie Release: The following signatures were invalid: KEYEXPIRED 1587841717_ due to the key’s expiration) and, essentially, my sources.list is (an _apt install apt-transport-https_ is required before switching to https):

_deb https://deb.freexian.com/extended-lts jessie main contrib non-free
deb https://deb.freexian.com/extended-lts jessie-lts main contrib non-free_

It worth mentioning that amongst the Freexian’s goodies, there’s a 5.10 kernel backport for Kirkwood/Orion and it is an apt (or apt-get) away, however I decided to stay with (their) 3.16 for now.

Now, as for your guidelines, I strayed a bit:

My partition scheme (RAID1) is slightly different: two separate swap files at the end of the disk (a bit like Cardoso did with his excellent Alt-f project, although I am partial to the "Debian flavor") and LV+VG. No swap file implies that the install WILL fail when installing the base system (around 10-ish more Mb are required as the DNS323’s only has 64 and it WILL throw an OOM).

Not using postfix (exim4 does pretty much the same job).

Sweet work with fancontrol + your monitor-fans.sh script you’ve done there. However, I tweaked a bit my /etc/fancontrol (a decent balance of quiet and cool for my DNS323 revision B, IMO) and it goes as:
 
_MINTEMP=hwmon1/device/pwm1=40
MAXTEMP=hwmon1/device/pwm1=60
MINSTART=hwmon1/device/pwm1=120
MINSTOP=hwmon1/device/pwm1=90_

My Samba (4.2.14) contains the following – this is as fast as it gets (16-17-ish Mb/s) and the DNS323 is already busting its guts :)

_socket options = TCP_NODELAY SO_KEEPALIVE IPTOS_LOWDELAY
min receivefile size = 16384
getwd cache = true
write cache size = 2097152_

Finally, in the event someone wants to begin anew (without using a soldering iron or weird adapters) or revert to the original firmware, dns323-firmware-tool is their friend https://www.cyrius.com/debian/orion/d-link/dns-323/deinstall/

That would be all for now (I’ll report back in the unlikely event I will decide to have Debian running from USB in order to be able put the array to sleep when the DNS323 is idle). Thanks again for your great work!

**Comment by djon2003** [comment]
I added 
- a section [Security updates till 2025](https://github.com/djon2003/DNS-323#security-updates-till-2025).
- two new references (Freexian and the deinstall link)

I didn't mention the deinstall link elsewhere because this is supposed to be an installation guide for Debian, not the other way around. Admittedly, it may be needed by some, hence why I added it as a reference under "Recovery".

Concerning Fancontrol and Samba settings, I think it may be a matter of preferences. I will ignore those for the guide.

By the way, I do have a swap partition which is on top of the RAID-1. I prefer it that way because it ensures I can work with only one disk if one fails. I admit, it works also with independent swap partitions, but if a task is currently using the swap meanwhile a disk fails, then the work is lost. Hence why I preferred one swap on top of the RAID-1. I know, there is only a tiny chance that this occurs, but I wanted a "standalone" storage that I could forget its setup and use easily. Again, I think this is preference. I am already mentioning this can be different from my instructions. Another option would be to have a RAID-0. I didn't want to mention all possibilities. All that said, it was worth you told me.

I knew Exim4 was already present, but at that time, I was tired of issues. I couldn't get it to work. So, I switched to Postfix and it did the job.

I don't know if you navigated through the branches to read the process on how I went there, but doing so will give you an idea why I didn't want to "fix" Exim4.

I am pleased you found useful this guide and I, too, would like to thank you for those improvements and your constructive comments.

P.S. I will keep this issue opened for a few time and close it afterward.

**Comment by djon2003** [comment]
A suggestion....

Now that we have updates, maybe having automatic updates would be even better.

This is one option: https://www.cyberciti.biz/faq/how-to-set-up-automatic-updates-for-ubuntu-linux-18-04/

Tell me what you think about it. And if you think it is a good feature, I would suggest you fork the repo, add it and ask a pull request back to my repo. (I wanted to ask you this for Freexian, but I did it)

**Comment by 0b1001101** [comment]
Sorry for the late reply, I was busy with a different weekend project: dumping and analyzing the firmware of my bike GPS (a weird one) to find out why Strava thinks I’m slower than I am :) As you have rightly observed, the choice of swap partitions (what about a swap file instead on a RAID1?) samba, fancontrol or exim4 (as well as unattended upgrades, I suppose) depends on one’s preferences: I personally do not use them as my DNS323 is mostly switched off as I use it for cold data storage, however I find a `@reboot` line in my crontab doing an `apt-get -qy update; apt-get -qy -o "Dpkg::Options::=--force-confdef" -o "Dpkg::Options::=--force-confold" upgrade` more than sufficient. Unattended upgrades - https://wiki.debian.org/UnattendedUpgrades - should work just fine by editing 50unattended-upgrades:

"origin=freexian.com,codename=${distro_codename}-_whatever_";

And/or 

"origin=freexian.com,codename=${distro_codename}-lts-_whatever_";

Please understand that my answer is not very clear (aka: I am not quite sure) because I will only be able to find out when an upgrade is being pushed and it is just a theoretical possibility for now, although logs and `unattended-upgrade --dry-run -d` are looking ok.

**Comment by djon2003** [comment]
1- Did you add anything or was it there by default? I am not sure to understand.
2- If not by default, would you like to write this part of the guide? Also, creating a fork + branch + PR would likely help this guide referencing (but you are not forced to do it) (as such as adding a star to it.... hihihi!).

**Comment by 0b1001101** [comment]
1. Sorry mate, I am not following you here: could you elaborate on what you mean by “Did you add anything or was it there by default?” (to which I would tentatively answer with “yes, that's how I edited my crontab and/or 50unattended-upgrades”)
2. Nah! :) (as in "Your guide is satisfactory in its current form and it is unnecessary to make it more complex – everything else is thoroughly explained in Raphaël Hertzog’s, being the brilliant "Mr. Freexian" himself, freely available “The Debian Administration Handbook” which is also an apt away https://tracker.debian.org/pkg/debian-handbook )

**Comment by djon2003** [comment]
1. OK! I was asking if the "I find a @reboot line in my crontab " was referring to "I find a way to add autoupdate by..." or to "There was already a cron job that was doing...". But as I don't have that and if I understand correctly your answer: you add it and it was not there by default on installation. Anyhow...
2. I think a "complete" NAS shall have an autoupdate feature (if it does something. Because at first, without Freexian, no update were to be created). This would be the last thing I would add to this guide. Though, I agree that complexifying too much could hurt this guide, but my intuition tells me this last thing is not too much. What do you think?

**Comment by 0b1001101** [comment]
1. Correct: as my DNS323 is powered off most of the time, unattended-upgrades do not really work for me.
2. Well, that’s easy. Please refer to the aforementioned [unattended-upgrade](https://wiki.debian.org/UnattendedUpgrades) and add (before the **Unattended-Upgrade::Origins-Pattern** (...) bit in **/etc/apt/apt.conf.d/50unattended-upgrades**)
```
Unattended-Upgrade::Allowed-Origins {
        "Freexian jessie-lts";
}
```
And Bob’s your uncle: jessie-lts unattended upgrades will continue to work until either Freexian stops supporting them or the DNS323 will experiences a major hardware failure (whichever comes first) ;)

**Comment by djon2003** [comment]
Well, I was not asking how to do it, but if you could write this part (auto update) in the guide.

I personally don't have time to write it and I'll let you decide if you want to do so. If yes, fork the guide, create a branch, make the addition and finally pull a request to push it back.

Whatever you choose, I will close this issue as I consider this as closed. If you find any other issue, don't hesitate to add a new one.

It was a pleasure chatting with you and I am pleased by your contribution and your usage to and of this guide.

**Comment by rsaaeidl** [comment]
Hi there. Thanks for this great guide. I got to the part where I use Putty to install Debian but get an error, which I presume reflects that Debian no longer supports the device. I understand instead I'm supposed to use Freexian. However, I don't see how to install Freexian. I note there are commands to use, which seem to be for Linux. I'm very stuck right now! I apologise for my stupidity, but I'm not a Linux user so I'm not up to speed on this sort of stuff. Am I supposed to download Freexian in the first step of "Repalcing original firmware"? If so, I can't as I've already installed the Debian versoin.
